homework-submission/week3/src/App.js

```diff
@@ -0,0 +1,60 @@
+import React, { Component } from "react";
+import "./App.css";
+import TodoApp from "./todo/TodoApp";
+import todos from "./todo/todos";
+
+class App extends Component {
+  state = {
+    todos
+  };
+
+  handelChange = id => {
+    const newTodos = this.state.todos.map(todo => {
+      const newTodo = {
+        ...todo,
+        done: todo.id === id ? !todo.done : todo.done
+      };
+      return newTodo;
+    });
+    this.setState({ todos: newTodos });
+  };
+
+  onAdd = value => {
+    const id = this.state.todos.length + 1;
+    value.id = id;
+    const newTodos = [...this.state.todos, value];
+    this.setState({
+      todos: newTodos
+    });
+  };
+
+  onUpdate = (id, description, deadline) => {
```

Comment:
Hmm, try to think about whether you're cloning the `this.state.todos`  or changing its values.
It comes to what we've talked about last time: `The Differences Between Primitives&References`. 

So, think about how to savely copy the `this.state.todos` object and THEN change some of its values.

Reply:
can you provide me with your way to hard copy the `this.state.todos` object ?

Reply:
Sure, take a look at this: https://devdocs.io/javascript/global_objects/object/assign

_HINT:_ Your `result` variable will be an object, which means you can assign it to a newly created object!!